### PR TITLE
bpo-1635741: Port the termios to multi-phase init (PEP 489)

### PR DESCRIPTION
**Pull request by koubaa**
https://bugs.python.org/issue1635741

### CODE REVIEW

**Comment by koubaa** [comment]
@vstinner @shihai1991 @corona10 please review

File: Modules/termios.c [hunk 1]
```diff
@@ -997,44 +1012,43 @@ static void termiosmodule_free(void *m) {
...
-    termiosmodulestate *state = get_termios_state(m);
+    termiosmodulestate *state = get_termios_state(mod);
     state->TermiosError = PyErr_NewException("termios.error", NULL, NULL);
     if (state->TermiosError == NULL) {
-        return NULL;
+        return -1;
     }
     Py_INCREF(state->TermiosError);
-    PyModule_AddObject(m, "error", state->TermiosError);
+    PyModule_AddObject(mod, "error", state->TermiosError);
```
Comment (vstinner) [inline Modules/termios.c]: Please add error handling (you need to DECREF in the error path). Same for the PyModule_AddIntConstant() call below.

File: Modules/termios.c [hunk 2]
```diff
@@ -82,28 +80,29 @@ static PyObject *
 termios_tcgetattr(PyObject *self, PyObject *args)
```
Comment (vstinner) [inline Modules/termios.c]: Would you mind to rename self parameter to module? Same remark for other functions that you modified to add a PyModule_GetState() call.

Comment (koubaa) [reply Modules/termios.c]: @vstinner sure, the CI is stuck anyways so I need to push something anyways

**Comment by vstinner** [comment]
Merged, thanks.

**Pull request merged**